components/script/dom/navigationpreloadmanager.rs

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::NavigationPreloadManagerBinding::NavigationPreloadState;
+use crate::dom::bindings::codegen::Bindings::NavigationPreloadManagerBinding::{
+    NavigationPreloadManagerMethods, Wrap,
+};
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject, Reflector};
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::bindings::str::ByteString;
+use crate::dom::domexception::{DOMErrorName, DOMException};
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::promise::Promise;
+use crate::dom::serviceworkerregistration::ServiceWorkerRegistration;
+use dom_struct::dom_struct;
+use js::jsval::UndefinedValue;
+use std::rc::Rc;
+
+#[dom_struct]
+pub struct NavigationPreloadManager {
+    reflector_: Reflector,
+    enabled: DomRefCell<bool>,
+    header_value: DomRefCell<Option<ByteString>>,
+    serviceworker_registration: Dom<ServiceWorkerRegistration>,
+}
+
+impl NavigationPreloadManager {
+    fn new_inherited(registration: &ServiceWorkerRegistration) -> NavigationPreloadManager {
+        NavigationPreloadManager {
+            reflector_: Reflector::new(),
+            enabled: DomRefCell::new(false),
+            header_value: DomRefCell::new(None),
```

Comment:
`enabled` and `header_value` should be members of `ServiceWorkerRegistration` instead of `NavigationPreloadManager`: https://w3c.github.io/ServiceWorker/#service-worker-registration-navigation-preload-enabled-flag

Reply:
Ah!! I see!! Per the spec, we'll save `enabled` and `header_value` in `ServiceWorkerRegistration` and get these values via `NavigationPreloadManager`! Will update the PR, thanks!